src/js/components/List/List.js

```diff
@@ -252,54 +245,41 @@ const List = React.forwardRef(
       ariaProps['aria-activedescendant'] = activeId;
     }
 
-    const onSelectOption = useCallback(
-      (event) => {
-        if ((onClickItem || onOrder) && active >= 0) {
-          if (onOrder) {
-            const index = Math.trunc(active / 2);
-            // Call onOrder with the re-ordered data.
-            // Update the active control index so that the
-            // active control will stay on the same item
-            // even though it moved up or down.
-            const newIndex = active % 2 ? index + 1 : index - 1;
-            onOrder(reorder(orderableData, pinnedInfo, index, newIndex));
-            updateActive(
-              active % 2
-                ? Math.min(active + 2, orderableData.length * 2 - 2)
-                : Math.max(active - 2, 1),
-            );
-          } else if (
-            disabledItems?.includes(getValue(data[active], active, itemKey))
-          ) {
-            event.preventDefault();
-          } else if (onClickItem) {
-            event.persist();
-            const adjustedEvent = event;
-            adjustedEvent.item = data[active];
-            adjustedEvent.index = active;
-            onClickItem(adjustedEvent);
-            sendAnalytics({
-              type: 'listItemClick',
-              element: listRef.current,
-              event: adjustedEvent,
-              item: data[active],
-              index: active,
-            });
-          }
+    const onSelectOption = (event) => {
```

Comment:
Pausing about removing `useCallback` vs just adding the needed dependencies. Reading docs about function caching, is there value to keeping it if it wasn't causing bugs?

https://react.dev/reference/react/useCallback